controllers/hbase_controller.go

```diff
@@ -628,11 +634,14 @@ const (
 )
 
 func (r *HBaseReconciler) statefulSet(hb *hbasev1.HBase,
-	stsName, cmName types.NamespacedName, ss hbasev1.ServerSpec) (*appsv1.StatefulSet, string) {
+	stsName, cmName types.NamespacedName, ss hbasev1.ServerSpec, rc int32) (*appsv1.StatefulSet, string) {
 	spec := (&ss.PodSpec).DeepCopy()
 	spec.Volumes = append(spec.Volumes, configMapVolume(cmName))
 	stsSpec := appsv1.StatefulSetSpec{
-		Replicas:            pointer.Int32Ptr(ss.Count),
+		// We don't want to use replica value as part of revision change (since
+		// scaling a statefulset is not the same as a new revision) so we just
+		// set current count for now just for hash calculation.
+		Replicas:            pointer.Int32Ptr(rc),
```

Comment:
why not just leave it empty?

Reply:
Because leaving it empty still forces a SHA changes the first time this code is deployed, causing all components to restart, even if it's just once and never again. From commit msg:

"When revision SHA is calculated, current replica count is used instead of some other arbitrary number. This is to avoid the first time this update is deployed triggering another full restart of all components (since arbitrary count would still be different than whatever the count it is currently the first time), but rather, make this change to SHA calculation entirely seamless."

Reply:
sounds good, sorry that didn't check the review commit